tools/system_libs.py

```diff
@@ -1665,6 +1640,10 @@ def add_library(lib):
   if shared.Settings.JS_MATH:
     libs_to_link = [(system_libs_map['libjsmath'].get_path(), True)] + libs_to_link
 
+  # When LINKABLE is set the whole link line is wrapped in --whole-archive
+  if shared.Settings.LINKABLE and shared.Settings.WASM_BACKEND:
+    return [l[0] for l in libs_to_link]
```

Comment:
If this was missing before, how and where was `LINKABLE` handled? If it was handled somewhere else, do we need to delete that?

Reply:
Yes, LINKABLE is handled already in `Building.link_lld`.. for some reason this extra nested use of `--whole-archive` was messing up this change in that case.     I can maybe split this part out?

Reply:
I split this out: https://github.com/emscripten-core/emscripten/pull/11120